x/staging/subspaces/keeper/common_functions.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/staging/subspaces/types"
+)
+
+// IterateSubspaces iterates through the subspaces set and performs the given function
+func (k Keeper) IterateSubspaces(ctx sdk.Context, fn func(index int64, subspace types.Subspace) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.SubspaceStorePrefix)
+	defer iterator.Close()
+
+	i := int64(0)
+	for ; iterator.Valid(); iterator.Next() {
+		var subspace types.Subspace
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &subspace)
+		stop := fn(i, subspace)
+		if stop {
+			break
+		}
+		i++
+	}
+}
+
+// checkSubspaceAndAdmin check if the subspaces with the given id exists and
+// if the address belongs to the admin of the subspace or one of its admins.
+// It returns an error or the subspace itself if everything's fine.
+func (k Keeper) checkSubspaceAndAdmin(ctx sdk.Context, id, address string) (types.Subspace, error) {
+	subspace, found := k.GetSubspace(ctx, id)
+	if !found {
+		return types.Subspace{}, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
+			"the subspace with id %s doesn't exist", id,
+		)
+	}
+
+	if subspace.Owner != address {
+		if !types.IsPresent(subspace.Admins, address) {
```

Comment:
Can we change this to be `subspace.IsAdmin(address)` instead? I think it would be easier to have it as a `Subspace` method instead of a function

Reply:
It's a method cause it's generic for both admins and registered/blocked users

Reply:
@bragaz I think we can make them different methods to be more explicit: 

```golang
func (s *Subspace) IsAdmin(user string) bool {
  return isPresent(s.Admins, user)
}

func (s *Subspace) IsBlocked(user string) bool {
  return isPresent(s.BlockedUsers, user)
}

func (s *Subspace) IsRegistered(user string) bool {
  return isPresent(s.RegisteredUsers, user)
}
```

This way we encapsulate different functionalities in different methods that all rely on the same private one (`isPresent`). We do not expose the underlying implementation but only the functionalities